service/storage_proxy.cc

```diff
@@ -134,7 +134,7 @@ static future<ResultTuple> encode_replica_exception_for_rpc(gms::feature_service
     std::exception_ptr eptr = f.get_exception();
     if (features.typed_errors_in_read_rpc) {
         if (auto ex = replica::try_encode_replica_exception(eptr); ex) {
-            return make_ready_future<ResultTuple>(utils::tuple_insert<ResultTuple>(SourceTuple{}, std::move(ex)));
+            return make_ready_future<ResultTuple>(utils::tuple_insert<ResultTuple>(utils::make_default_rpc_tuple<SourceTuple>(), std::move(ex)));
```

Comment:
Given this is an exceptional path, my comment about optimizing it isn't relevant.